backend/v3/storage/database/events_testing/settings_instance_test.go

```diff
@@ -0,0 +1,1790 @@
+						settingsRepo.OwnerTypeCondition(domain.OwnerTypeInstance),
+					),
+				),
+			)
+			require.NoError(t, err)
+
+			// event instance.policy.login.multifactor.added
+			assert.Equal(t, []domain.MultiFactorType{domain.MultiFactorType(policy.MultiFactorType_MULTI_FACTOR_TYPE_U2F_WITH_VERIFICATION)}, setting.MFAType)
+			assert.WithinRange(t, *setting.UpdatedAt, before, after)
+		}, retryDuration, tick)
+	})
+
+	t.Run("test added/removed second multifactor reduces", func(t *testing.T) {
+		ctx := t.Context()
+		before := time.Now()
+		newInstance := integration.NewInstance(t.Context())
+
+		IAMCTX := newInstance.WithAuthorization(ctx, integration.UserTypeIAMOwner)
+
+		// get current second factor types
+		var secondFactorTypes []domain.SecondFactorType
+		retryDuration, tick := integration.WaitForAndTickWithMaxDuration(ctx, time.Second*20)
+		assert.EventuallyWithT(t, func(t *assert.CollectT) {
+			setting, err := settingsRepo.Get(
+				ctx, pool,
+				database.WithCondition(
+					database.And(
+						settingsRepo.InstanceIDCondition(newInstance.ID()),
+						settingsRepo.OrganizationIDCondition(nil),
+						settingsRepo.TypeCondition(domain.SettingTypeLogin),
+						settingsRepo.OwnerTypeCondition(domain.OwnerTypeInstance),
+					),
+				),
+			)
+			require.NoError(t, err)
+
+			secondFactorTypes = setting.SecondFactorTypes
+		}, retryDuration, tick)
+
+		// add new second factor type
+		before = time.Now()
+		_, err := newInstance.Client.Admin.AddSecondFactorToLoginPolicy(IAMCTX, &admin.AddSecondFactorToLoginPolicyRequest{
+			Type: policy.SecondFactorType_SECOND_FACTOR_TYPE_OTP_SMS,
+		})
+		require.NoError(t, err)
+		after := time.Now()
+
+		secondFactorTypes = append(secondFactorTypes, domain.SecondFactorType(policy.SecondFactorType_SECOND_FACTOR_TYPE_OTP_SMS))
+
+		// check new second factor type is added
+		retryDuration, tick = integration.WaitForAndTickWithMaxDuration(ctx, time.Second*20)
+		assert.EventuallyWithT(t, func(t *assert.CollectT) {
+			setting, err := settingsRepo.Get(
+				ctx, pool,
+				database.WithCondition(
+					database.And(
+						settingsRepo.InstanceIDCondition(newInstance.ID()),
+						settingsRepo.OrganizationIDCondition(nil),
+						settingsRepo.TypeCondition(domain.SettingTypeLogin),
+						settingsRepo.OwnerTypeCondition(domain.OwnerTypeInstance),
+					),
+				),
+			)
+			require.NoError(t, err)
+
+			// event instance.policy.login.multifactor.secondfactor.added
+			assert.Equal(t, secondFactorTypes, setting.SecondFactorTypes)
+			assert.WithinRange(t, *setting.UpdatedAt, before, after)
+		}, retryDuration, tick)
+
+		// remove second factor type
+		before = time.Now()
+		_, err = newInstance.Client.Admin.RemoveSecondFactorFromLoginPolicy(IAMCTX, &admin.RemoveSecondFactorFromLoginPolicyRequest{
+			Type: policy.SecondFactorType_SECOND_FACTOR_TYPE_OTP_SMS,
+		})
+		require.NoError(t, err)
+		after = time.Now()
+
+		secondFactorTypes = secondFactorTypes[0 : len(secondFactorTypes)-1]
```

Comment:
Remove the second factor type with slices function, can change depending on sorting.